lib/builders.ncl

```diff
@@ -84,4 +93,75 @@ in
         structured_env.buildInputs = packages,
       }
         | (NickelPkg & { packages | { _ : Derivation } }),
+
+  ShellApplication
+    | doc m%"
+    "%
+    =
+      {
+        name,
+        version | default = "0.0.0",
+        text | NullOr NixString | MutExclusiveWith file "text" "file" | default = null,
+        file | NullOr String | default = null,
+        runtime_inputs | { _ : Derivation } = {},
+        shell_binary | NixString | default = nix-s%"%{lib.import_nix "nixpkgs#bash"}/bin/bash"%,
+
+        build_command = {
+          cmd = nix-s%"%{lib.import_nix "nixpkgs#bash"}/bin/bash"%,
+          args = [
+            "-c",
+            nix-s%"
+              set -euo pipefail
+              source .attrs.sh
+              PATH="%{lib.import_nix "nixpkgs#coreutils"}/bin:$PATH"
+              mkdir -p "${outputs[out]}/bin"
+              target="${outputs[out]}/bin/${name}"
+              %{if text != null then
+                m%"cat > "$target" <<<"$script""%
+                else
+                m%"(cat <<<"$scriptHeader"; cat "$scriptPath") > "$target""%
+              }
+              chmod +x "$target"
+              "%{shell_binary}" -n -O extglob "$target"
+              "%{lib.import_nix "nixpkgs#shellcheck"}/bin/shellcheck" "$target"
+            "%
+          ],
+        },
+        nix_drv =
```

Comment:
```suggestion
        env =
```

Reply:
`env` is handled by `NixpkgsPkg`, this is `NickelPkg`.

Reply:
Oh indeed. Why?

Reply:
`stdenv` doesn't provide much benefit here, so why depend on it if we can avoid it?

Reply:
Well, it's easier and more standard. So I'd rather use it everywhere for the sake of simplicity